# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/nsICookieJarSettings.idl
# Commit: f605a7ddf950
# Full Hash: f605a7ddf9502fe74187f430417e9686d10d4b96
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2023-03-03 09:56:45
# Regressor Bug: 1816064
# File Overlap Count: 1
# Description:
#   Bug 1816064 - Part 2: Implement the fingerprinting randomization key to cookieJarSettings. r=tjr,pbz
#   
#   This patch adds a fingerprintingRandomizationKey to the
#   nsICookieJarSettings. The random key will be generated when loading a
#   top-level http channel and set to the cookieJarSettings. The key will
# ==============================================================================

diff -r 68074ac5a154 -r f605a7ddf950 netwerk/cookie/nsICookieJarSettings.idl
--- a/netwerk/cookie/nsICookieJarSettings.idl	Thu Mar 02 21:47:06 2023 +0000
+++ b/netwerk/cookie/nsICookieJarSettings.idl	Thu Mar 02 21:47:06 2023 +0000
@@ -60,6 +60,11 @@
   readonly attribute AString partitionKey;
 
   /**
+   * The key used for fingerprinting randomization.
+   */
+  readonly attribute Array<uint8_t> fingerprintingRandomizationKey;
+
+  /**
    * CookiePermission at the loading of the document for a particular
    * principal. It returns the same cookiePermission also in case it changes
    * during the life-time of the top document.